main.go

```diff
@@ -2,48 +2,58 @@ package main
 
 import (
 	"context"
-	_ "net/http/pprof"
+	"flag"
+	"io/ioutil"
+	"log"
 	"os"
 
 	logging "github.com/ipfs/go-log/v2"
 
 	"github.com/filecoin-project/venus/cmd"
 )
 
+var (
+	logLevel string
+)
+
+func init() {
+	flag.StringVar(&logLevel, "log-level", "", "Set log level (default: info)")
+}
+
 func main() {
+	flag.Parse()
+
 	// set default log level if no flags given
-	lvl := os.Getenv("GO_FILECOIN_LOG_LEVEL")
-	if lvl == "" {
+	if logLevel == "" {
 		logging.SetAllLoggers(logging.LevelInfo)
-		_ = logging.SetLogLevel("beacon", "error")
-		_ = logging.SetLogLevel("peer-tracker", "error")
-		_ = logging.SetLogLevel("dht", "error")
-		_ = logging.SetLogLevel("bitswap", "error")
-		_ = logging.SetLogLevel("graphsync", "info")
-		_ = logging.SetLogLevel("heartbeat", "error")
-		_ = logging.SetLogLevel("dagservice", "error")
-		_ = logging.SetLogLevel("peerqueue", "error")
-		_ = logging.SetLogLevel("swarm", "error")
-		_ = logging.SetLogLevel("swarm2", "error")
-		_ = logging.SetLogLevel("basichost", "error")
-		_ = logging.SetLogLevel("dht_net", "error")
-		_ = logging.SetLogLevel("pubsub", "error")
-		_ = logging.SetLogLevel("relay", "error")
-		_ = logging.SetLogLevel("dht/RtRefreshManager", "error")
 	} else {
-		level, err := logging.LevelFromString(lvl)
+		level, err := logging.LevelFromString(logLevel)
 		if err != nil {
-			level = logging.LevelInfo
+			log.Fatalf("Error setting log level: %v", err)
 		}
 		logging.SetAllLoggers(level)
 	}
 
-	if len(os.Args) > 1 {
-		if os.Args[1] == "-v" {
-			os.Args[1] = "version"
+	// read logger configuration file if present
+	if _, err := os.Stat("logger.json"); !os.IsNotExist(err) {
+		file, err := ioutil.ReadFile("logger.json")
+		if err != nil {
+			log.Fatalf("Error reading logger configuration file: %v", err)
 		}
+		if err := logging.SetLogLevelConfigString(string(file)); err != nil {
```

Comment:
undefined: log.SetLogLevelConfigString

cycles call SetLogLevel after the JSON parsing; or call SetupLogging？

Reply:
The correct function to set the logger configuration is `logging.SetLogLevelConfigString`. So, after reading the logger configuration file, the program should call `logging.SetLogLevelConfigString` to set the logger configuration.

Reply:
This should be called after setting the log level based on the command-line flag, and before calling cmd.Run. The logging.SetLogLevelConfigString function parses the logger configuration JSON string and sets the log levels for the loggers defined in the configuration. It should be called after setting the log level because it may override the log level set by the command-line flag.